packages/e2e-tests/src/puppeteerEnvironment.js

```diff
@@ -39,15 +40,42 @@ class PuppeteerEnvironment extends OriginalEnvironment {
 
   async handleTestEvent(event, state) {
     if (event.name === 'test_fn_failure') {
-      const testName = state.currentlyRunningTest.name;
-      await this.storeArtifacts(testName);
+      const testName = `${state.currentlyRunningTest.parent.name}  ${state.currentlyRunningTest.name}`;
+      const errors = state.currentlyRunningTest.errors;
+      const eventError = util.inspect(event);
+      let errorMessages = '';
+      errorMessages += `========= ${testName} ==========\n\n`;
+      errorMessages +=
+        'started:' +
+        new Date(event.test.startedAt).toLocaleString() +
+        ' ended:' +
+        new Date().toLocaleString();
+      errorMessages += '============end==========\n\n';
+      errors.forEach((error) => {
+        errorMessages += `${testName}:${error}\n\n`;
+      });
+
+      errorMessages += '=========================\n\n';
+      errorMessages += eventError;
+
+      await this.storeArtifacts(testName, errorMessages);
+
+      if (eventError.includes('JestAssertionError')) {
+        // return already handled
+        return;
+      }
+
+      // eslint-disable-next-line no-console
+      console.log(' Unhandled event(' + event.name + '): ' + eventError);
```

Comment:
Hmm now this leads to duplicate output:

https://github.com/GoogleForCreators/web-stories-wp/runs/7356729173?check_suite_focus=true#step:8:86

Maybe not needed anymore?

Reply:
Removed ---it's in the artifacts logs as needed.